homeassistant/components/nederlandse_spoorwegen/sensor.py

```diff
@@ -37,13 +43,25 @@
 ROUTES_SCHEMA = vol.All(cv.ensure_list, [ROUTE_SCHEMA])
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {vol.Required(CONF_API_KEY): cv.string, vol.Optional(CONF_ROUTES): ROUTES_SCHEMA}
+    {
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Optional(CONF_EMAIL): cv.string,
+        vol.Optional(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_ROUTES): ROUTES_SCHEMA,
+    }
 )
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the departure sensor."""
 
+    if config.get(CONF_EMAIL) or config.get(CONF_PASSWORD):
```

Comment:
I don't see the point of this since the integration won't work if the api key isn't set in the config.

Reply:
Makes sense… so then the appropriate way of dealing with such a breaking change is to make the integration crash and make people look up the documentation on their own?

Reply:
The config validation will stop the integration from being set up if the config is incorrect. The log error will notify the user what's wrong with the config.

Reply:
If we would have had more time, we could have deprecated the old config options during a transition period. That's not an option now since the api has already changed.